src/plugins/visualizations/public/vis.ts

```diff
@@ -200,7 +200,7 @@ export class Vis<TVisParams = VisParams> {
       data: {
         aggs: aggs as any,
         searchSource: this.data.searchSource ? this.data.searchSource.getSerializedFields() : {},
-        savedSearchId: this.data.savedSearchId,
+        ...(this.data.savedSearchId ? { savedSearchId: this.data.savedSearchId } : {}),
```

Comment:
When adding a by reference Visualize panel, the `savedSearchId` key does not exist in `this.data`. However, when unlinking, this key was injected in to `this.data` via this original line - i.e. `this.data.savedSearchId = undefined | string`. Unfortunately, our current deep comparison of panels when saving treats a key not existing and `key = undefined` as unequal - this prevented the `unsaved changes` badge from being cleared.

To prevent this, the `savedSearchID` key only gets added if it has a value.